testing/cpm/cpm_proprietary-url-ctk-version-find-ctk.cmake

```diff
@@ -0,0 +1,50 @@
+#=============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include(${rapids-cmake-dir}/cpm/init.cmake)
+include(${rapids-cmake-dir}/cpm/package_override.cmake)
+include(${rapids-cmake-dir}/cpm/detail/get_proprietary_binary_url.cmake)
+include(${rapids-cmake-dir}/cpm/detail/package_details.cmake)
+
+rapids_cpm_init()
+
+# Need to write out an override file with a proprietary blob url
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/override.json
+  [=[
+{
+  "packages" : {
+    "test_binary" : {
+      "version" : "2.6.1",
+      "proprietary_binary" : {
+        "x86_64-linux" :  "https://fake.url.com/${version}/${cuda-toolkit-version}/x86_64_${cuda-toolkit-version-major}.tgz",
```

Comment:
I assume the answer is no since it's kind of the whole point of bracket expressions, but just in case: is there any way to force variable evaluation inside brackets? It would be nice to not need to repeat the URL every time.

Reply:
You mean the fact that we have both x86 and arm? It is an explicit choice so that we can explicitly state support for multiple platforms in a easy to understand manner.

To have only a single url we would need:
- Introduce a key named `all_platforms`
- Add extra keys / values to  `proprietary_binary` that mapped from the system arch ( x86 / arm ) to the placeholders used by that package in the download url

Reply:
No, I just mean the repetition of the domain in the URL `https://fake.url.com`.

Reply:
That is a good idea. Want to create an issue for that and we can iterate on it independent of this PR?

Reply:
https://github.com/rapidsai/rapids-cmake/issues/379